### ISSUE TITLE
Make Notification Factory cache certificates, or make the factory implement an interface

### ISSUE THREAD

**Issue by stevenmyhre** [mac-cain13/notificato#55]
I get "bad password" errors when I send multiple notifications in the same process - using a new message builder for each one, creating a new certificate for each (via ```Notificato::createCertificate``` method).  Simple solution was to cache the certificate objects so they can be used again.  But it would be nice if the ```CertificateFactory``` supported this.  It only caches a default one.  I'd like to key the cache off of the pem location.  Was going to implement my own CertificateFactory for this, but turns out the ```setCertificateFactory``` typehints to the concrete implementation of ```Apns\CertificateFactory```.  I can make my own factory and use it outside the scope of this library, but thought I would recommend it to the repo... sorry, no time for PRs!

**Comment by mac-cain13** [comment]
Agreed, nice idea! I don't have the time and incentive to implement this right now. But leave it open until somebody has the time to implement this in the library.
